jax/_src/config.py

```diff
@@ -295,33 +324,40 @@ def _trace_context(self):
     return (self.x64_enabled, self.jax_numpy_rank_promotion,
             self.jax_default_matmul_precision)
 
-class _BoolStateContextManager:
-  def __init__(self, name, help, update_thread_local_hook):
+class _StateContextManager:
+  def __init__(self, name, help, update_thread_local_hook,
+               validate_new_val_hook: Callable[[Any], None] = None):
     self._name = name
     self.__name__ = name[4:] if name.startswith('jax_') else name
     self.__doc__ = f"Context manager for `{name}` config option.\n\n{help}"
-    self._hook = update_thread_local_hook
+    self._update_thread_local_hook = update_thread_local_hook
+    self._validate_new_val_hook = validate_new_val_hook
 
   @contextlib.contextmanager
-  def __call__(self, new_val: bool):
+  def __call__(self, new_val):
+    if self._validate_new_val_hook:
```

Comment:
Do you want `is not None` for consistency with the other lines?

(Actually I prefer the way this line is written, since we know it's either a callable or None. Whatever floats your boat though, I just wanted to flag the inconsistency!)

Reply:
Good catch. I changed the others to not have `is not None` for brevity.